### ISSUE TITLE
NaN gradients at the identity: quaternion_to_axis_angle and axis_angle_to_quaternion differentiate through an unguarded sqrt(0)

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#3949]
`quaternion_to_axis_angle` and `axis_angle_to_quaternion` both differentiate through an unguarded `sqrt(0)` at the identity rotation, so backward returns **NaN** at the single most common input in the domain: the identity quaternion `(1, 0, 0, 0)` and the zero axis-angle `(0, 0, 0)`.

```python
>>> q = torch.tensor([1., 0., 0., 0.], requires_grad=True); quaternion_to_axis_angle(q).sum().backward(); q.grad
tensor([nan, nan, nan, nan])
>>> v = torch.zeros(3, requires_grad=True); axis_angle_to_quaternion(v).sum().backward(); v.grad
tensor([nan, nan, nan])
```

The two sources:

```python
# quaternion_to_axis_angle, kornia/geometry/conversions.py:758-760
sin_squared_theta: torch.Tensor = q1 * q1 + q2 * q2 + q3 * q3
sin_theta: torch.Tensor = torch.sqrt(sin_squared_theta)

# axis_angle_to_quaternion, kornia/geometry/conversions.py:896-898
theta_squared: torch.Tensor = a0 * a0 + a1 * a1 + a2 * a2
theta: torch.Tensor = torch.sqrt(theta_squared)
```

In both cases the forward pass is fine — a subsequent `torch.where(… > 0.0, k_pos, k_neg)` selects the correct branch, so the returned *value* at the singularity is right. But `torch.where` propagates gradients through **both** arms, and `d/dx sqrt(x)` at `x = 0` is `inf`, so `inf * 0 = NaN` poisons the whole gradient. The fix pattern is already in this very file: `axis_angle_to_rotation_matrix` writes `torch.sqrt(theta2.clamp(min=1e-12))  # clamping to ensure no nan gradients` and returns a finite gradient at the same point.

## Repro

```python
import torch
from kornia.geometry.conversions import (
    axis_angle_to_quaternion as aa2q,
    axis_angle_to_rotation_matrix as aa2R,
    quaternion_to_axis_angle as q2aa,
)

q = torch.tensor([1.0, 0.0, 0.0, 0.0], dtype=torch.float64, requires_grad=True)
q2aa(q).sum().backward()
print("q2aa grad at the identity quaternion (1,0,0,0):", q.grad)

v = torch.zeros(3, dtype=torch.float64, requires_grad=True)
aa2q(v).sum().backward()
print("aa2q grad at the zero axis-angle (0,0,0)      :", v.grad)

# the same file already has the fix pattern:
w = torch.zeros(1, 3, dtype=torch.float64, requires_grad=True)
aa2R(w).sum().backward()
print("aa2R grad at the zero axis-angle (theta2.clamp):", w.grad)

# both are finite away from the singularity
q2 = torch.tensor([0.9, 0.1, 0.2, 0.3], dtype=torch.float64, requires_grad=True)
q2aa(q2).sum().backward()
print("q2aa grad at (0.9,0.1,0.2,0.3)                :", q2.grad)
```

## Observed

```
q2aa grad at the identity quaternion (1,0,0,0): tensor([nan, nan, nan, nan], dtype=torch.float64)
aa2q grad at the zero axis-angle (0,0,0)      : tensor([nan, nan, nan], dtype=torch.float64)
aa2R grad at the zero axis-angle (theta2.clamp): tensor([[0., 0., 0.]], dtype=torch.float64)
q2aa grad at (0.9,0.1,0.2,0.3)                : tensor([-1.2632,  2.0155,  1.9249,  1.8344], dtype=torch.float64)
```

## Expected

A finite gradient at the identity, as `axis_angle_to_rotation_matrix` already produces on the third line above. The limit is well-defined for both maps: as `theta → 0`, `axis_angle_to_quaternion` approaches `q = (1, v/2)` and `quaternion_to_axis_angle` approaches `v = 2·(x, y, z)`, both smooth, both with bounded derivatives — there is no mathematical singularity here, only a parametrization artifact of computing `theta = sqrt(theta²)` and then dividing by it.

Why this matters more than most numerical warts: these are the differentiable-library core paths. Any pose-optimization, camera-calibration, or rotation-regression loop that initializes at (or passes through) the identity rotation — which is the overwhelmingly common initialization — gets NaN in the first backward pass and the whole optimization dies. The failure is silent in forward and total in backward.

The fix is the same one line already used elsewhere in the file: clamp the radicand before the `sqrt` (`torch.sqrt(x.clamp(min=eps))` with `eps` at or below `finfo.tiny`), or use a double-`where` so the `sqrt` never sees zero on either arm. Both leave the forward values byte-identical away from the singularity, and produce the correct finite limit at it.

Prior art: #2924 (closed) reported exactly this failure mode for `axis_angle_to_rotation_matrix`; the `theta2.clamp(min=1e-12)` there is the accepted fix. The two functions in this issue were never given the same treatment. #1653 (closed) and #1267 (closed) are the analogous reports on the matrix→quaternion and matrix→axis-angle paths.

## Classification

**Broken output (NaN).** Per the reference-core lifecycle (Part I / Part III §2), clearly-broken-output bugs — NaN, crashes, tautological checks — may land early as ordinary bugfixes without waiting for the coordinated Repair & Deprecation window, because brokenness has no users to protect. The forward path is unaffected, so the repair can be landed under #3916-style byte-identity discipline on all non-degenerate default paths.

## Environment

kornia 0.9.0rc1 @ `3358ef16`, torch 2.9.1, cpu. Executed from a worktree carrying the batch-3a docs/tests commits on top of `3358ef16`; `quaternion_to_axis_angle` and `axis_angle_to_quaternion` are byte-identical to `origin/main` there (no diff hunk lands in either; function bodies compare AST-identical).

Posted on behalf of @ducha-aiki by Claude (Fable 5).
